src/fireEvent.js

```diff
@@ -9,9 +9,15 @@ const findEventHandler = (
 ) => {
   const eventHandler = toEventHandlerName(eventName);
 
-  if (typeof element.props[eventHandler] === 'function') {
+  if (
+    typeof element.props[eventHandler] === 'function' &&
+    element.props.disabled !== true
```

Comment:
This will work fine in most use cases, but since we check not only host components, it may prevent firing events on custom components that happen to have prop named `disabled` that doesn't prevent events.